### DISCUSSION TITLE
Current implementation uses the key name of the transfered asset as id for the secrets associated with the transfer.

### DISCUSSION THREAD

**Discussion by FlorianSeidelAWS** [eclipse-edc/Connector#2417] [Ideas]
I noticed, that an assets key name is used for naming secrets used in the transfer process. Since there is no guarantee, that this is going to be unique in the future, this can cause problems. May I propose to replace this with the transfer process id? I think it is available in all places in the code where secrets are stored, deleted an retrieved for transfers.

**Comment by jimmarino** [comment]
Which ones are you specifically referring to?

**Comment by FlorianSeidelAWS** [comment]
TransferProcessManager::storeProvisionedSecrets() stores secret using key name of provisioned resource
MultiPartArtifactRequestSender retrieves secret using key name
S3DataSinkFactory uses key name to retrieve secret 
There must be other places, too.

**Reply by jimmarino** [reply]
That is by design. Keys for accessing assets are associated with the data address, not the transfer process.
